### ISSUE TITLE
Clarification on domOverlay element and sizing

### ISSUE THREAD

**Issue by RafaelCintron** [immersive-web/dom-overlays#18]
The spec has the following to say about domOverlay element sizing:

>While active, the DOM overlay element is automatically resized to fill the dimensions of the UA-provided DOM overlay rectangle. Its background color is automatically styled as transparent for the duration of the session.

In "Example 2", the code fetches the `uiElement` directly from the DOM and passes it to `requestSession`

```JS
let uiElement = document.getElementById('ui');
navigator.xr.requestSession('immersive-ar', {
    optionalFeatures: ['dom-overlay'],
    domOverlay: { root: uiElement } }).then((session) => {
    ...
  }
}
```

For scenarios where the 2D web page is rendered on a separate monitor, how does sizing and transparency affect how the element appears in the 2D web page?  

Automatically making element backgrounds be transparent could have security implications for domOverlay elements on the 2D page that are in front of other elements in different security domains.

**Comment by klausw** [comment]
As far as I know, it would be technically challenging for UAs to show the same DOM element simultaneously on two separate output devices, especially if they had different display styling, and may even be impossible, so the UA would likely need to stop displaying the overlay element on the 2D screen while it's being used as an active overlay in the session.

Should this be required behavior to make things consistent across platforms? Something along these lines on top of https://github.com/immersive-web/dom-overlays/pull/17/files ?

> On a multi-display system where the immersive session uses a separate output device from the originally displayed web page, such as a headset attached to a desktop PC, the DOM overlay element MUST NOT be visible on other displays while it is being shown in the immersive view. UAs MUST treat it as if it had `display: none` styling on those other displays, and MUST NOT generate DOM input events for interactions using input devices associated with these non-immersive displays.

**Comment by klausw** [comment]
... For the last sentence, I meant "MUST NOT generate DOM input events for interactions with the overlay element" - this wasn't intended to imply that all input to the 2D page needs to be blocked.

Would it make sense to give UAs additional flexibility by adding an option that a UA may just entirely stop displaying the 2D page during the session, i.e. showing a "session in progress" placeholder instead? This may be significantly easier to implement than dealing with detached DOM elements.

**Comment by thetuvix** [comment]
It seems reasonable to me to say that UAs MUST NOT show the element on the original page and also that they MAY choose to hide the whole page.

**Comment by klausw** [comment]
@thetuvix, I'm adding another sentence in https://github.com/immersive-web/dom-overlays/pull/19 to cover the "hide the whole page" option. The first part was already included in the current text at https://immersive-web.github.io/dom-overlays/#fullscreen-api-integration . What do you think?

**Comment by RafaelCintron** [comment]
If we prevent user agents from showing the main page, we eliminate 'demo driver' as a scenario in desktop VR.  

In demo driver, one person wears the VR headset while another one sits at the computer and uses the main page to "drive" what the VR headset wearing person sees.  The main page can have options like "Number of enemies", "current score", "Reset scene", etc.  Based on the headset wearing person's experience, the demo driver can adjust it in real time. 

I agree with @thetuvix's language suggestion of "MAY choose to hide the whole page".

**Comment by klausw** [comment]
PR #19 proposes adding: _The UA MAY choose to hide or disable the entire page on other displays for the duration of the session._ Does that match what you have in mind?

I agree that the "demo driver" scenario would be worth preserving as an option, but it would be up to the UA if that's possible or not.

**Comment by RafaelCintron** [comment]
@klausw , yes #19 would satisfy my concerns.

**Comment by klausw** [comment]
Thanks! I merged #19, closing.
